packages/ui/src/council/modals/AnnounceCandidacy/AnnounceCandidacyModal.tsx

```diff
@@ -263,25 +263,36 @@ export const AnnounceCandidacyModal = () => {
 
   if (state.matches('announceCandidacyTransaction')) {
     return (
-      <AnnounceCandidacyTransaction
-        onClose={hideModal}
+      <SignTransactionModal
+        buttonText="Sign transaction and Announce"
+        textContent={
+          <>
+            <TextMedium>You intend to announce candidacy.</TextMedium>
+            <TextMedium>
+              Also you intend to stake <TokenValue value={form.watch('staking.amount') ?? BN_ZERO} />.
+            </TextMedium>
+          </>
+        }
         transaction={announceCandidacyTransaction}
         signer={activeMember.controllerAccount}
-        stake={form.watch('staking.amount') ?? BN_ZERO}
+        onClose={hideModal}
         service={state.children.announceCandidacyTransaction}
```

Comment:
```suggestion
```
Can't these 2 always be accessed from inside `SignTransactionModal` with:
```ts
  const { modal, hideModal, currentModalMachine, showModal, modalData } = useModal()
  const [service] = useMachine(currentModalMachine)
```
?

Reply:
I guess `onClose` can be removed, but service names differ from modal to modal.